src/coffe/spice.py

```diff
@@ -202,8 +204,304 @@ def run(self, sp_path: str, parameter_dict: Dict[str, List[str]]):
+        pair is a list of different parameter values that you want to run HSPICE on.
+        The lists must be of the same length for all params in 'parameter_dict' (param1_name's
+        list has the same number of elements as param2_name). Here's what is going to happen:
+        We will start by setting all the parameters to their 'val1' and we'll run NGSPICE. 
+        Then, we'll set all the params to 'val2' and we'll run NGSPICE. And so on (that's 
+        actually not exactly what happens, but you get the idea). So, you can run NGSPICE on 
+        different transistor size conbinations by adding elements to these lists. Transistor 
+        sizing combination i is ALL the vali in the parameter_dict. So, even if you never 
+        want to change param1_name, you still need a list (who's elements will all be the 
+        same in this case).
+        If you only want to run NGSPICE for one transistor sizing combination, your lists will
+        only have one element (but it still needs to be a list).
+
+        Ok, so 'parameter_dict' contains a key-value pair for each transistor where the 'value'
+        is a list of transistor sizes to use. A transistor sizing combination consists of all 
+        the elements at a particular index in these lists. You also need to provide a key-value
+        (or key-list we could say) for all your wire RC parameters. The wire RC data in the 
+        lists corresponds to each transistor sizing combination. You'll need to calculate what
+        the wire RC is for a particular transistor sizing combination outside this function 
+        though. Here, we'll only set the paramters to the appropriate values. 
+
+        Finally, what we'll return is a dictionary similar to 'parameter_dict' but containing
+        all of the of the SPICE measurements. The return value will have this format: 
+
+        measurements = {meas_name1: [value1, value2, value3, etc...], 
+                        meas_name2: [value1, value2, value3, etc...],
+                        etc...}
+        """
+        sp_dir = os.path.dirname(sp_path)
+        sp_filename = os.path.basename(sp_path)
+
+        numOfValues = len(next(iter(parameter_dict.values())))
+
+        # modify the wire subcircuit so the syntax is NGSPICE compatible
+        basic_subcircuit_path = "basic_subcircuits.l"
+        basic_subcircuit_file = open(basic_subcircuit_path, "r+")
+        basic_subcircuit_content = basic_subcircuit_file.read()
+        # Rw and Cw become {Rw} and {Cw}
+        basic_subcircuit_content_modified = re.sub(r"\b([RC]w)$", r"{\1}", basic_subcircuit_content, flags=re.M|re.I)
+        basic_subcircuit_file.seek(0)
+        basic_subcircuit_file.write(basic_subcircuit_content_modified)
+        basic_subcircuit_file.truncate()
+        basic_subcircuit_file.close()
+
+        sp_file = open(os.path.join(sp_dir, sp_filename), "r+")
+        sp_content = sp_file.read()
+        # functions used by regex subsititution
```

Comment:
I suggest adding a comment with a list of the changes to the .sp file needed by ngspice. You kind of have that in the comments below, but giving a concise summary would be helpful.